packages/medusa/src/services/order.ts

```diff
@@ -1907,10 +1910,10 @@ class OrderService extends TransactionBaseService {
         }
       }
 
-      if (receivedReturn.refund_amount > 0) {
+      if (refundAmount > 0) {
```

Comment:
self-review: In case you pass a custom refund amount e.g. `500` upon receiving a return, the previous implementation would ignore it and refund the amount from the return.